cpp/src/Slice/Grammar.y

```diff
@@ -584,94 +605,14 @@ type_id
 // ----------------------------------------------------------------------
 optional
 // ----------------------------------------------------------------------
-: ICE_OPTIONAL_OPEN ICE_INTEGER_LITERAL ')'
+: ICE_OPTIONAL_OPEN integer_constant ')'
```

Comment:
We used to have 2 separate grammar rules for matching `ICE_INTEGER_LITERAL` and `scoped_name`.
We did this in multiple places, so I created an intermediate grammar rule `integer_constant` which handles both.